ckanext/harvest/tests/nose/test_action.py

```diff
@@ -258,6 +259,73 @@ def test_create(self):
         assert 'url' in result
         assert u'There already is a Harvest Source for this URL' in result['url'][0]
 
+class TestHarvestSourceActionList(FunctionalTestBase):
+
+    def test_list_with_organization(self):
+        organization = ckan_factories.Organization.create()
+        harvest_data = {
+            "owner_org": organization["id"],
+            "type": "harvest",
+            "url": "https://www.gov.uk/random",
+            "source_type": "test-nose"
+        }
+
+        other_harvest_data = {
+            "type": "harvest",
+            "url": "https://www.gov.uk/other-path",
+            "source_type": "test-nose"
+        }
+
+        harvest = ckan_factories.Dataset.create(**harvest_data)
+        harvest_source = factories.HarvestSource.create(id=harvest["id"])
+
+        other_harvest = ckan_factories.Dataset.create(**other_harvest_data)
+        other_harvest_source = factories.HarvestSource.create(id=other_harvest["id"])
+
+        sysadmin = ckan_factories.Sysadmin()
+
+        app = _get_test_app()
+        response = app.get('/api/action/{0}'.format('harvest_source_list'),
+                           params={"organization_id":organization["id"]},
+                           status=200)
+
+
+        results = response.json['result']
+
+        result_harvest = model.Session.query(model.Package).get(results[0]["id"])
+        result_organization_id = result_harvest.owner_org
+
+        assert response.json['success'] is True
+        assert 1 is len(results)
+        assert_equal(organization["id"], result_organization_id)
+
+    def test_list_without_organization_and_limit(self):
```

Comment:
Can we also have a test where there is no limit set, as this could be the case if other devs don't set the HARVEST_SOURCE_LIST_LIMIT config

Reply:
Sure, I've added a test and re-organised them slightly so all test cases should be covered